PostHog/Autocapture/PostHogAutocaptureIntegration.swift

```diff
@@ -0,0 +1,468 @@
+    }
+
+    extension UIControl {
+        @objc override func ph_shouldTrack(_ action: Selector, for target: Any?) -> Bool {
+            actions(forTarget: target, forControlEvent: ph_autocaptureEvents)?.contains(action.description) ?? false
+        }
+    }
+
+    extension UIScrollView {
+        override var ph_autocaptureDebounceInterval: TimeInterval { 0.4 }
+    }
+
+    extension UISegmentedControl {
+        override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+        override var ph_autocaptureText: String? { titleForSegment(at: selectedSegmentIndex) }
+    }
+
+    extension UIPageControl {
+        override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+    }
+
+    extension UISearchBar {
+        override var ph_autocaptureEvents: UIControl.Event { .editingDidEnd }
+    }
+
+    extension UIToolbar {
+        override var ph_autocaptureEvents: UIControl.Event {
+            if #available(iOS 14.0, *) { .menuActionTriggered } else { .primaryActionTriggered }
+        }
+    }
+
+    extension UITextField {
+        override var ph_autocaptureText: String? { text ?? attributedText?.string ?? placeholder }
+    }
+
+    extension UITextView {
+        override var ph_autocaptureText: String? { text ?? attributedText?.string }
+    }
+
+    extension UIStepper {
+        override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+        override var ph_autocaptureText: String? { "\(value)" }
+    }
+
+    extension UISlider {
+        override var ph_autocaptureDebounceInterval: TimeInterval { 0.3 }
+        override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+        override var ph_autocaptureText: String? { "\(value)" }
+    }
+
+    extension UISwitch {
+        @objc override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+        override var ph_autocaptureText: String? { "\(isOn)" }
+    }
+
+    extension UIPickerView {
+        override var ph_autocaptureText: String? {
+            (0 ..< numberOfComponents).reduce("") { result, component in
+                let selectedRow = selectedRow(inComponent: component)
+                if let title = delegate?.pickerView?(self, titleForRow: selectedRow, forComponent: component) {
+                    return result.isEmpty ? title : "\(result) \(title)"
+                }
+                if let title = delegate?.pickerView?(self, attributedTitleForRow: selectedRow, forComponent: component) {
+                    return result.isEmpty ? title.string : "\(result) \(title.string)"
+                }
+                return result
+            }
+        }
+    }
+
+    #if !os(tvOS)
+        extension UIDatePicker {
+            override var ph_autocaptureEvents: UIControl.Event { .valueChanged }
+        }
+    #endif
+
+    private func sanitizeText(_ title: String) -> String {
+        title.replacingOccurrences(of: "[^a-zA-Z0-9.]+", with: "-", options: .regularExpression, range: nil)
+    }
```

Comment:
Maybe we use this that already exists in the SDK?:

```
extension String {
    func mask() -> String {
        String(repeating: "*", count: count)
    }
}
```

Reply:
Mm, yeah, I'm not sure why that was done. I thought it was masking, but it's sanitizing special characters and spaces. I did not pay attention to it. What does the JS SDK do?

Reply:
I see, your call then, either like js-lite or js is is fine since both are working and nobody complained.